DMCompiler/DM/Visitors/DMProcBuilder.cs

```diff
@@ -634,6 +636,24 @@ public void ProcessStatementForType(DMASTProcStatementForType statementForType)
             _proc.OutputControl();
         }
 
+        public void ProcessStatementOutput(DMASTProcStatementOutput statementOutput) {
+            DMExpression.Emit(_dmObject, _proc, statementOutput.Right);
+            DMExpression left = DMExpression.Create(_dmObject, _proc, statementOutput.Left);
+            (DMReference leftRef, _) = left.EmitReference(_dmObject, _proc);
+
+            _proc.Output(leftRef);
```

Comment:
Requiring the left side of an output to be a reference makes statements like `list(world.log) << "Hello, World!"` impossible. There should be two versions of the `Output` opcode, one that uses a reference and one that uses two DreamValues.

The one that uses two DreamValues can perform the `Output` on each value in a list, or in statements like `file("output.txt") << "Message"` instead of the current hack with `GlobalProc` references. It can fall back to doing nothing otherwise.